File: src/particle.cpp
#include "particle.h"
#include "istime.h"

void particle_render(Particle &p) {
	glColor4f(p.color.x, p.color.y, p.color.z,
			1.0f - (glfwGetTime() - p.begin) / (p.end - p.begin));
	glVertex3f(p.pos.x, p.pos.y, p.pos.z);
	p.pos = vec3_lincom(p.pos, 1, p.vel, getDelta());
}

File: Docking/src/com/pi/phys/Particle.java
package com.pi.phys;

import org.lwjgl.opengl.GL11;

import com.pi.Main;
import com.pi.math.Vector3;

public class Particle {
	Vector3 color;
	Vector3 pos;
	Vector3 vel;
	double begin, end;

	public Particle() {
		color = new Vector3();
		pos = new Vector3();
		vel = new Vector3();
		begin = end = 0;
	}

	public void render() {
//		GL11.glColor3f(color.x, color.y, color.z,
//				1.0f - (float) ((Main.getTime() - begin) / (end - begin)));
		// It is always black >.>
		GL11.glVertex3f(pos.x, pos.y, pos.z);
		Vector3.addto(pos, vel, (float) Main.getDelta());
	}
}
